runtime/near-vm-runner/src/logic/logic.rs

```diff
@@ -3168,14 +3127,11 @@ bls12381_p2_decompress_base + bls12381_p2_decompress_element * num_elements`
             .into());
         }
         self.result_state.gas_counter.pay_per(storage_read_key_byte, key.len() as u64)?;
-        let nodes_before = self.ext.get_trie_nodes_count();
-        let read = self.ext.storage_get(&key, self.config.storage_get_mode);
```

Comment:
Subtle: unlike many other places in the previous code, this *does* charge fees regardless of whether the operation has been successful or not.

`write`/`remove` will bail before recording the TTN fees if reading the value failed.